pyscript.core/docs/README.md

```diff
@@ -0,0 +1,481 @@
+
+`@pyscript/core` adds a functionality called `XWorker` to all of the interpreters it offers, which works in each language the way `Worker` does in JavaScript.
+
+In each Interpreter, `XWorker` is a global reference, with a counter `xworker` (lower case) global reference within the worker code.
+
+In short, the `XWorker` global goal is to help, without much thinking, to run any desired interpreter out of a *Worker*, enabling extra features on the *worker*'s code side.
+
+### XWorker options
+
+Before showing any example, it's important to understand how the offered API differs from Web standard *workers*:
+
+| name      | example                                                  | behavior |
+| :-------- | :------------------------------------------------------- | :--------|
+| async     | `XWorker('./file.py', async=True)`                       | The worker code is evaluated via `runAsync` utility where, if the *interpreter* allows it, top level *await* would be possible, among other *PL* specific asynchronous features.  |
+| config    | `XWorker('./file.py', config='./cfg.toml')`              | The worker will load and parse the *JSON* or *TOML* file to configure itself. Please see [currently supported config values](https://docs.pyscript.net/latest/reference/elements/py-config.html#supported-configuration-values) as this is currently based on `<py-config>` features. |
+| type      | `XWorker('./file.py', type='pyodide')`                   | Define the *interpreter* to use with this worker which is, by default, the same one used within the running code. Please read the [Terminology](#terminology) **interpreter** dedicated details to know more. |
+| version   | `XWorker('./file.py', type='pyodide', version='0.23.2')` | Allow the usage of a specific version where, if numeric, must be available through the project *CDN* used by *core* but if specified as fully qualified *URL*, allows usage of any interpreter's version: `<script type="pyodide" version="http://localhost:8080/pyodide.local.mjs">` |
+
+The returning *JS* reference to any `XWorker(...)` call is literally a `Worker` instance that, among its default API, have the extra following feature:
+
+
+| name      | example                            | behavior |
+| :-------- | :--------------------------------- | :--------|
+| sync      | `sync = XWorker('./file.py').sync` | Allows exposure of callbacks that can be run synchronously from the worker file, even if the defined callback is *asynchronous*. This property is also available in the global `xworker` reference. |
+
+```python
+
+sync = XWorker('./file.py').sync
```

Comment:
Think that using `sync` as variable name here can be more confusing to the end user.... it can become confusing in regards of what's part of the API and what's just the variable name